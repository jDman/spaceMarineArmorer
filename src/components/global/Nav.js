import React from 'react';

import { NavLink } from 'react-router-dom';

function Header() {
  return (
    <nav>
      <ul>
        <li>
          <NavLink to="/">Home</NavLink>
        </li>
        <li>
          <NavLink to="/admin">Admin</NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Header;
