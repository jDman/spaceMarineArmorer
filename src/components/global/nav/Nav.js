import React, { Fragment } from 'react';

import { NavLink } from 'react-router-dom';

import Button from '../../button/Button';
import classes from './Nav.module.css';

function Nav(props) {
  const activeLingStyles = {
    color: '#0900DB',
  };

  return (
    <nav>
      <ul className={classes.Navigation}>
        {props.isAuth ? (
          <li className={classes.NavigationListItem}>
            <NavLink
              to={'/'}
              className={classes.NavigationLink}
              exact
              strict
              activeStyle={activeLingStyles}
            >
              Home
            </NavLink>
          </li>
        ) : null}

        {!props.isAuth && (
          <Fragment>
            <li className={classes.NavigationListItem}>
              <NavLink
                to="/signup"
                className={classes.NavigationLink}
                exact
                strict
                activeStyle={activeLingStyles}
              >
                Signup
              </NavLink>
            </li>
            <li className={classes.NavigationListItem}>
              <NavLink
                to="/login"
                className={classes.NavigationLink}
                exact
                strict
                activeStyle={activeLingStyles}
              >
                Login
              </NavLink>
            </li>
          </Fragment>
        )}

        {props.isAuth && (
          <Fragment>
            {props.isAdmin ? (
              <li className={classes.NavigationListItem}>
                <NavLink
                  to="/admin"
                  className={classes.NavigationLink}
                  exact
                  strict
                  activeStyle={activeLingStyles}
                >
                  Admin
                </NavLink>
              </li>
            ) : null}

            <li className={classes.NavigationListItem}>
              <NavLink
                to="/shop"
                className={classes.NavigationLink}
                exact
                strict
                activeStyle={activeLingStyles}
              >
                Shop
              </NavLink>
            </li>

            <li className={classes.NavigationListItem}>
              <NavLink
                to="/cart"
                className={classes.NavigationLink}
                exact
                strict
                activeStyle={activeLingStyles}
              >
                Cart
              </NavLink>
            </li>

            <li className={classes.NavigationListItem}>
              <Button
                classes={classes.NavigationLogoutButton}
                clickHandler={() => props.logout()}
                buttonText="Logout"
              />
            </li>
          </Fragment>
        )}
      </ul>
    </nav>
  );
}

export default Nav;
