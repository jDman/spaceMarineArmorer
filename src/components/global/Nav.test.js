import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Nav from './Nav';

describe('<Nav />', () => {
  it('renders Nav component without isAuth props', () => {
    const component = renderer.create(
      <Router>
        <Nav />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Nav component with isAuth props', () => {
    const component = renderer.create(
      <Router>
        <Nav isAuth={true} />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
