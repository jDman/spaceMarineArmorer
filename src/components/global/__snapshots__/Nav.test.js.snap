// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders Nav component with isAuth props 1`] = `
<nav>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </li>
  </ul>
</nav>
`;

exports[`<Nav /> renders Nav component without isAuth props 1`] = `
<nav>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/signup"
        onClick={[Function]}
      >
        Signup
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
  </ul>
</nav>
`;
